Signal/src/ViewControllers/ContactsViewHelper.m

```diff
@@ -0,0 +1,261 @@
+    self.contactAccountMap = [contactAccountMap copy];
+
+    [self.delegate contactsViewHelperDidUpdateContacts];
+}
+
+- (BOOL)isContactHidden:(Contact *)contact
+{
+    OWSAssert([NSThread isMainThread]);
+
+    if (contact.parsedPhoneNumbers.count < 1) {
+        // Hide contacts without any valid phone numbers.
+        return YES;
+    }
+
+    if ([self.delegate shouldHideLocalNumber] && [self isCurrentUserContact:contact]) {
+        // We never want to add ourselves to a group.
+        return YES;
+    }
+
+    return NO;
+}
+
+- (BOOL)isCurrentUserContact:(Contact *)contact
+{
+    OWSAssert([NSThread isMainThread]);
+
+    for (PhoneNumber *phoneNumber in contact.parsedPhoneNumbers) {
+        if ([[phoneNumber toE164] isEqualToString:[TSAccountManager localNumber]]) {
+            return YES;
+        }
+    }
+
+    return NO;
+}
+
+- (NSString *)localNumber
+{
+    return [TSAccountManager localNumber];
+}
+
+- (BOOL)isContactBlocked:(Contact *)contact
+{
+    OWSAssert([NSThread isMainThread]);
+
+    if (contact.parsedPhoneNumbers.count < 1) {
+        // Do not consider contacts without any valid phone numbers to be blocked.
+        return NO;
+    }
+
+    for (PhoneNumber *phoneNumber in contact.parsedPhoneNumbers) {
+        if ([_blockedPhoneNumbers containsObject:phoneNumber.toE164]) {
+            return YES;
+        }
+    }
+
+    return NO;
+}
+
+- (BOOL)isRecipientIdBlocked:(NSString *)recipientId
+{
+    AssertIsOnMainThread();
+
+    return [_blockedPhoneNumbers containsObject:recipientId];
+}
+
+- (NSArray<Contact *> *_Nonnull)filteredContacts
+{
+    AssertIsOnMainThread();
+
+    NSMutableArray<Contact *> *result = [NSMutableArray new];
+    for (Contact *contact in self.contactsManager.signalContacts) {
+        if (![self isContactHidden:contact]) {
+            [result addObject:contact];
+        }
+    }
+    return [result copy];
+}
+
+- (BOOL)doesContactAccount:(ContactAccount *)contactAccount matchSearchTerm:(NSString *)searchTerm
+{
+    OWSAssert(contactAccount);
+    OWSAssert(searchTerm.length > 0);
+
+    if ([contactAccount.contact.fullName.lowercaseString containsString:searchTerm.lowercaseString]) {
+        return YES;
+    }
+
+    NSString *asPhoneNumber = [PhoneNumber removeFormattingCharacters:searchTerm];
+    if (asPhoneNumber.length > 0 && [contactAccount.recipientId containsString:asPhoneNumber]) {
+        return YES;
+    }
+
+    return NO;
+}
+
+- (BOOL)doesContactAccount:(ContactAccount *)contactAccount matchSearchTerms:(NSArray<NSString *> *)searchTerms
+{
+    OWSAssert(contactAccount);
+    OWSAssert(searchTerms.count > 0);
+
+    for (NSString *searchTerm in searchTerms) {
+        if (![self doesContactAccount:contactAccount matchSearchTerm:searchTerm]) {
+            return NO;
+        }
+    }
+
+    return YES;
+}
+
+- (NSArray<ContactAccount *> *)contactAccountsMatchingSearchString:(NSString *)searchText
```

Comment:
`OWSContactsSearcher` assumed that there was a single search term.  I'm hoping to remove `OWSContactsSearcher`.